lab_share_lib/processing/base_processor.py

```diff
@@ -1,7 +1,17 @@
-from typing import Callable
+from abc import ABC, abstractmethod
+from typing import Any
 
 from lab_share_lib.processing.rabbit_message import RabbitMessage
+from lab_share_lib.rabbit.basic_publisher import BasicPublisher
+from lab_share_lib.rabbit.schema_registry import SchemaRegistry
 
 
-class BaseProcessor:
-    process: Callable[["BaseProcessor", RabbitMessage], bool]
+class BaseProcessor(ABC):
+    @staticmethod
+    @abstractmethod
+    def instantiate(
+        schema_registry: SchemaRegistry, basic_publisher: BasicPublisher, config: Any
+    ) -> "BaseProcessor": ...
```

Comment:
This definition enforces an abstract static instantiate method in message processors. 
Example1MessageProcessor in examples/example1/processors.py does not implement it.  It would fail in lab_share_lib/processing/rabbit_message_processor.py while building the message processor. I think the consumer example is broken.

Reply:
This is a good catch.  I felt like there was something lissing in my updates to the example, but I couldn't put my finger on it.  I've replaced the `init` method with the required `instantiate` method after declaring `BaseProcessor` as the super class.